### DISCUSSION TITLE
Aplicação MEAN com NGINX como proxy reverso

### DISCUSSION THREAD

**Discussion by mahenrique94** [frontendbr/forum#701] [Perguntas e Respostas]
Fala ai galera, seguinte, tenho um pequeno problema.

Tenho uma aplicação feita com **MEAN** que esta funcionando corretamente, porém ao realizar o deploy da mesmo dentro de um container no docker e utilizando o **NGINX** como *proxy reverso*, ela não encontra nenhum arquivo `.js`, segue anexos:

Sem NGINX como proxy reverso:
![Sem Proxy](http://res.cloudinary.com/mahenrique94/image/upload/v1504196380/Screen_Shot_2017-08-31_at_13.11.11_ut9y3k.png)

Com NGINX como proxy reverso:
![Com Proxy](http://res.cloudinary.com/mahenrique94/image/upload/v1504196380/Screen_Shot_2017-08-31_at_13.10.06_roph3p.png)

Dentro desse NGINX tenho várias outras aplicações feitas em **JAVA** e **PHP**, onde funcionam corretamente, apenas essa com MEAN estou tendo esse problema.

Abraçooos

**Comment by rsmelo** [comment]
ele está tentando puxar os arquivos de http://app.atenainformatica.com.br, ele não está adicionando o /doccetus no seu  index onde você importa os arquivos.

**Comment by RatoX** [comment]
Cara, cola tua configuração do Nginx ai

**Comment by RatoX** [comment]
@mahenrique94 ?

**Comment by mahenrique94** [comment]
Fala ai galera, desculpa a demora, acabei não vendo a respostas de vocês.

@rsmelo Realmente não estava adicionado o /doccetus, então sempre que faço o build do projeto eu tenho que ir no `index.html` e adicionar em todos os arquivos `.js` e `favico` o **doccetus/NOME_ARQUIVO**.

@RatoX segue gist do `nginx.conf`: [nginx.conf](https://gist.github.com/mahenrique94/c729a6caf5ed111a7926c1256ff8d78a)

Por hora consegui fazer o app subir e puxar os arquivos `.js`, tive que mexer na configuração do Express, segue gist do arquivo: [express](https://gist.github.com/mahenrique94/eeebba2cebd2ee39eee3cd545ce4920b)(a linha que adicionei esta comentada)

Mas ainda estou com problemas nos ícones e algumas comunicações com o Node, não encontrei nada no angular para fazer o build do `index.html` e uma pasta e os **assets** em outra pasta.

Abraçoooos

**Comment by feliperodrigues64** [comment]
Tem como eu fazer o ICE (Stun/Turn) pelo Nginx para o Node usar nas conexões P2P de webRTC? Tipo, eu queria segregar a sinalização ICE, que seria realizada pelo Node.JS para ser realizada peloNgInx, e depois repassar os dados e metadados para o Nede.Js fazer o P2P.  Seria algo como deixar as sinalizações isoladas no NgInx sempre que posível e o Node ficaria mais livre para tratar requisições http e para interpretar JavScript do lado do servidor.

**Comment by felipefialho** [comment]
@suissa

**Comment by feliperodrigues64** [comment]
Valeu LFeh! O Suissa é o cara! kkkkkk
